build_tools/index_generation_s3_tar.py

```diff
@@ -0,0 +1,241 @@
+                const ul = document.getElementById('fileList');
+                ul.innerHTML = '';
+                fileList.forEach(file => {{
+                    const li = document.createElement('li');
+                    const href = encodeURIComponent(file.name).replace(/%2F/g, '/');
+                    li.innerHTML = `<a href="${{href}}" target="_blank" rel="noopener noreferrer">${{file.name}}</a>`;
+                    ul.appendChild(li);
+                }});
+            }}
+            function updateDisplay() {{
+                const order = document.getElementById('sortOrder').value;
+                const filter = document.getElementById('filter').value;
+                let sortedFiles = [...files].sort((a, b) => {{
+                    return (order === 'desc') ? b.mtime - a.mtime : a.mtime - b.mtime;
+                }});
+                sortedFiles = applyFilter(sortedFiles, filter);
+                renderFiles(sortedFiles);
+            }}
+            document.addEventListener('DOMContentLoaded', function() {{
+                updateDisplay();
+                document.getElementById('sortOrder').addEventListener('change', updateDisplay);
+                document.getElementById('filter').addEventListener('change', updateDisplay);
+            }});
+        </script>
+    </head>
+    <body>
+        <h1>{page_title}</h1>
+        <div class="controls">
+            <label for="sortOrder">Sort by:</label>
+            <select id="sortOrder">
+                <option value="desc">Last Updated (Recent to Old)</option>
+                <option value="asc">First Updated (Old to Recent)</option>
+            </select>
+            <label for="filter">Filter by:</label>
+            <select id="filter">
+                <option value="all">All</option>
+                {gpu_families_options}
+            </select>
+        </div>
+        <ul id="fileList"></ul>
+    </body>
+    </html>
+    """
+
+    # Write locally
+    local_path = "index.html"
+    with open(local_path, "w", encoding="utf-8") as f:
+        f.write(html_content)
+
+    message = f"index.html generated successfully for bucket '{bucket_name}'. File saved as {local_path}"
+    gha_append_step_summary(message)
+    # Upload to bucket root
+    if upload:
+        try:
+            s3_client.upload_file(
+                local_path,
+                bucket_name,
+                "index.html",
+                ExtraArgs={"ContentType": "text/html"},
+            )
+
+            # URL to the uploaded index.html
+            region = s3_client.meta.region_name or "us-east-2"
+            if region == "us-east-2":
+                bucket_url = f"https://{bucket_name}.s3.amazonaws.com/index.html"
+            else:
+                bucket_url = (
+                    f"https://{bucket_name}.s3.{region}.amazonaws.com/index.html"
+                )
+
+            message = f"index.html successfully uploaded. URL: {bucket_url}"
```

Comment:
I would like to keep this, but can we make the log message better? Maybe:

```suggestion
            message = f"Index page uploaded at {bucket_url}"
```